/*
 * @begin:license
 *
 * Copyright (c) 2015-2019, Benjamin Niemann <pink@odahoda.de>
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * @end:license
 */

syntax = "proto2";

package noisicaa.pb;

import "noisicaa/node_db/node_description.proto";
import "noisicaa/audioproc/public/plugin_state.proto";

message PluginInstanceSpec {
  required string realm = 1;
  required string node_id = 2;
  required NodeDescription node_description = 3;
  optional PluginState initial_state = 4;
}

message CreatePluginRequest {
  required PluginInstanceSpec spec = 1;
  optional string callback_address = 2;
}

message CreatePluginResponse {
  required string pipe_path = 1;
}

message DeletePluginRequest {
  required string realm = 1;
  required string node_id = 2;
}

message CreatePluginUIRequest {
  required string realm = 1;
  required string node_id = 2;
}

message CreatePluginUIResponse {
  required uint32 wid = 1;
  required uint32 width = 2;
  required uint32 height = 3;
}

message DeletePluginUIRequest {
  required string realm = 1;
  required string node_id = 2;
}

message SetPluginStateRequest {
  required string realm = 1;
  required string node_id = 2;
  required PluginState state = 3;
}
